lib/ipmi_strings.c

```diff
@@ -1724,7 +1724,7 @@ int ipmi_oem_info_init()
 	oem_valstr_list_t terminator = { { -1, NULL}, NULL }; /* Terminator */
 	oem_valstr_list_t *oemlist = &terminator;
 	bool free_strings = true;
-	size_t count;
+	int count;
```

Comment:
Although I don't agree with the solution in general, this is a good catch anyway as `oem_info_list_load()` may return a negative value.